CONFIG.md

```diff
@@ -0,0 +1,52 @@
+# Configuration
+
+##### 10/14/19
+We've begun the migration to a new configuration framework.  The 'old' method (the `sip-communicator.properties` file and program arguments) will be supported throughout the transition and the bridge will be fully backwards-compatible with the old config.  No changes are necessary until we pull the plug on the old config completely, and we'll give ample warning before that time.
+
+Moving forward, we'll be using lightbend's [config library]([https://github.com/lightbend/config](https://github.com/lightbend/config)).  Their GitHub page gives lots of details, but below are the changes most relevant to configuring the JVB.
+
+### Configuration sources
+During the transition, we'll continue to read from the original locations (`sip-communicator.properties` and program arguments) for all configuration properties, falling back to the new `reference.conf` file for defaults.  The `reference.conf` file holds defaults for all configuration properties, and shouldn't be modified in a deployment.  Instead, users can put a file named `application.conf` in the classpath: values can be put there to override ones in `reference.conf`.  The lightbend docs do a good job of describing [how configuration is loaded]([https://github.com/lightbend/config#standard-behavior](https://github.com/lightbend/config#standard-behavior)).  Here's a more concrete example of overriding a configuration value for JVB:
```

Comment:
Doesn't `reference.conf` typically live inside a .jar file? If this file is intended to be user-facing rather than developer-facing it seems like mentioning it here at all might be confusing.

Reply:
not sure what you mean by 'user facing', but i did want to mention this for developers since it's where the defaults live.  maybe i can make that clearer

Reply:
I mean, the primary audience for this file should be people deploying jvb and writing config files, and for them the exact mechanism by which defaults are defined isn't terribly relevant.

If you want to have a separate section for developers, that's fine, but better to keep them separate.

Reply:
ok, that's seems reasonable. will move to a dev section

Reply:
actually, thinking about this again i do think it's useful to mention this.  if you were deploying the old code and wanted to know whether or not you had to override a property, you'd probably look up its default value.  the default values used to live in the code, but now people will need to look elsewhere to find them, so i don't think it's useful, even for users just deploying, to be able to find those defaults.

Reply:
The easiest way to find them is presumably to look at the logs at startup time, no? Rather than scraping the jar files.

Reply:
hm, i was picturing someone going to the repo.  because of the json representation in the log it isn't the easiest to browse.  ok, i'll try and come up with a tweak here

Reply:
hm, also, this would require running the bridge with your config cleared just to see all the default values.